### DISCUSSION TITLE
JSON parse error for $persist in incognito mode

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3523] [7. Bugs (archived)]
Inside `Alpine.data('$', function(){ return {` I'm using `view: this.$persist('raster')`. Now here is what I know which make me think I found a bug: The error only occurrs, when opening the site in (Google Chrome) incognito mode. When I clear my localstorage and open it in normal mode, it works as expected. What I don't get is where the `undefined` comes from...

Here is the error message:

> 5968:1 Uncaught SyntaxError: "undefined" is not valid JSON
    at JSON.parse (<anonymous>)
    at storageGet (alpine-persist.js:42:15)
    at Alpine.interceptor.func.as.alias (alpine-persist.js:8:51)
    at Object.initialize (alpine.js:351:14)
    at alpine.js:335:26
    at Array.forEach (<anonymous>)
    at recurse (alpine.js:330:59)
    at initInterceptors (alpine.js:343:10)
    at alpine.js:2818:3
    at Function.<anonymous> (alpine.js:1188:55)

**Comment by simonferndriger** [comment]
I somehow must have looked/changed the wrong version it seems to work now, never mind 🙈

**Comment by ekwoka** [comment]
Not sure the cause ,but just want to note for anyone that ends up here: you should sure both your `alpinejs` and any core plugins are at the same version number.

This is basically the only compatibility validation that is done.

Technically many other versions may work perfectly fine, but no such compatibility can be assumed, so make sure they're the same version whenever a plugin might be having an issue.
